src/Preprocessor.zig

```diff
@@ -560,14 +559,19 @@ fn expr(pp: *Preprocessor, tokenizer: *Tokenizer) MacroError!bool {
             .arrow,
             .period,
             => {
-                const tok = pp.tokens.get(start + i);
                 try pp.comp.diag.add(.{
                     .tag = .invalid_preproc_operator,
                     .loc = tok.loc,
                 }, tok.expansionSlice());
                 return false;
             },
             else => if (id.isMacroIdentifier()) {
+                try pp.comp.diag.add(.{
+                    .tag = .undefined_macro,
+                    .loc = tok.loc,
+                    .extra = .{ .str = pp.expandedSlice(tok) },
```

Comment:
Is it safe to use `pp.expandedSlice(tok)` here, or do I need to dupe it into comp.diag's arena allocator?

Reply:
`generated_buf` was moved to `Compilation` so it should live as long as `Diagnostics`.